tests/infer/test_valid_models.py

```diff
@@ -826,3 +827,55 @@ def model():
 
     with pyro.validation_enabled():
         assert_ok(model, model, TraceEnum_ELBO(max_iarange_nesting=1))
+
+
+@pytest.mark.parametrize("times", [1, 2], ids=["applied_once", "applied_twice"])
+def test_iarange_shape_broadcasting(times):
+    data = torch.ones(1000, 2)
+
+    def model():
+        with pyro.iarange("num_particles", 10, dim=-3):
+            with pyro.iarange("components", 2, dim=-1):
+                p = pyro.sample("p", dist.Beta(torch.tensor(1.1), torch.tensor(1.1)))
+                assert p.shape == torch.Size((10, 1, 2))
+            with pyro.iarange("data", data.shape[0], dim=-2):
+                pyro.sample("obs", dist.Bernoulli(p), obs=data)
+
+    def guide():
+        with pyro.iarange("num_particles", 10, dim=-3):
+            with pyro.iarange("components", 2, dim=-1):
+                pyro.sample("p", dist.Beta(torch.tensor(1.1), torch.tensor(1.1)))
```

Comment:
Maybe put this or something similar in the `handlers.broadcast` docstring as a usage example?

Reply:
Will update.